hack/generate-operator-bundle.py

```diff
@@ -4,153 +4,107 @@
-    prev_csv = sys.argv[8]
-
-    operator_assets_dir = os.path.join("deploy")
-    # Check to see if the deploy directory exists before going on.
-    if not os.path.exists(operator_assets_dir):
-        print >> sys.stderr, "ERROR Operator asset directory {} does not exist. Giving up.".format(operator_assets_dir)
-        sys.exit(1)
-
-    version_dir = os.path.join(operator_dir, operator_version)
-    if os.path.exists(version_dir):
-        print >> sys.stderr, "INFO version already exists, skipping: {}".format(version_dir)
-        sys.exit(0)
-
-    # doesn't exist, create the target version
+
+# This script will append the current number of commits given as an arg
+# (presumably since some past base tag), and the git hash arg for a final
+# version like: 0.1.189-3f73a592
+VERSION_BASE = "0.1"
+
+OPERATOR_NAME="must-gather-operator"
+
+if len(sys.argv) != 6:
+    print("USAGE: %s OUTPUT_DIR PREVIOUS_VERSION GIT_NUM_COMMITS GIT_HASH HIVE_IMAGE" % sys.argv[0])
+    sys.exit(1)
+
+outdir = sys.argv[1]
+prev_version = sys.argv[2]
+git_num_commits = sys.argv[3]
+git_hash = sys.argv[4]
+operator_image = sys.argv[5]
+
+full_version = "%s.%s-%s" % (VERSION_BASE, git_num_commits, git_hash)
+print("Generating CSV for version: %s" % full_version)
+
+if not os.path.exists(outdir):
+    os.mkdir(outdir)
+
+version_dir = os.path.join(outdir, full_version)
+if not os.path.exists(version_dir):
     os.mkdir(version_dir)
 
+with open('config/templates/{}-csv-template.yaml'.format(OPERATOR_NAME), 'r') as stream:
+    csv = yaml.load(stream)
+
+csv['spec']['customresourcedefinitions']['owned'] = []
+
+    # Copy all CSV files over to the bundle output dir:
+    # Copy all CRD files over to the bundle output dir:
+crd_files = [ f for f in os.listdir('deploy/crds') if f.endswith('_crd.yaml') ]
+for file_name in crd_files:
+    full_path = os.path.join('deploy/crds', file_name)
+    if (os.path.isfile(os.path.join('deploy/crds', file_name))):
+        shutil.copy(full_path, os.path.join(version_dir, file_name))
+    # Load CRD so we can use attributes from it
+    with open("deploy/crds/{}".format(file_name), "r") as stream:
+        crd = yaml.load(stream)
+    # Update CSV template customresourcedefinitions key
+    csv['spec']['customresourcedefinitions']['owned'].append(
+        {
+            "name": crd["metadata"]["name"],
+            "description": crd["spec"]["names"]["kind"],
+            "displayName": crd["spec"]["names"]["kind"],
+            "kind": crd["spec"]["names"]["kind"],
+            "version": crd["spec"]["version"]
+        }
+    )
+
+csv['spec']['install']['spec']['clusterPermissions'] = []
+
+# Add operator role to the CSV:
+with open('deploy/02_must-gather-operator.ClusterRole.yaml', 'r') as stream:
+    operator_role = yaml.load(stream)
+    csv['spec']['install']['spec']['clusterPermissions'].append(
+        {
+            'rules': operator_role['rules'],
+            'serviceAccountName': OPERATOR_NAME,
```

Comment:
I see what this is doing, but how does the SA in 04_must-gather-admin.ServiceAccount.yaml get created?

Reply:
My understanding is that OLM ensures that the SA is created ahead of time: https://github.com/operator-framework/olm-book/blob/master/docs/operator-scoping.md#scoping-member-operator-permissions